### DISCUSSION TITLE
로컬에 서버를 올리기 전 npm install 에러

### DISCUSSION THREAD

**Discussion by Yenowme** [devrel-kr/HackaLearn#48] [2 Q&A - 클라우드 스킬 챌린지] [answered]
과제에 적힌 지시를 따라 my-static-web-app레포를 클론 해 vue-app디렉토리에서 npm install 을 진행하는데 
<img width="1122" alt="스크린샷 2021-08-04 오후 6 24 11" src="https://user-images.githubusercontent.com/76904042/128156903-28c67102-8c6d-42e2-90b8-b90afa18b2a2.png">
여기서 install이 멈추더니 
<img width="967" alt="스크린샷 2021-08-04 오후 6 26 24" src="https://user-images.githubusercontent.com/76904042/128157252-9b34957c-b10b-4703-af80-664b42b982ef.png">
[2021-08-04T09_26_04_970Z-debug.log](https://github.com/devrel-kr/HackaLearn/files/6930176/2021-08-04T09_26_04_970Z-debug.log)

이런 에러가 뜹니다 (자세한 에러로그도 같이 첨부했습니다)
혹시 이유를 아시는 분 계실까요..?

**Comment by kooljay82** [comment]
vue-app 쪽에 이슈가 있어서 비슷한 오류인것 같았는데 아니네요~ npm install이 행해진 디렉토리가 어딘가요??

**Reply by Yenowme** [reply]
my-static-web-app/vue-app 에서 진행하고있습니다 ㅜㅜ mac os 빅서 사용중에있습니다 혹시나 해서 react-app 디렉토리에서 진행해봤는데도 동일한 에러가 뜨네요

**Reply by kooljay82** [reply]
node랑 npm 업데이트 하시고 package-lock.json 삭제 해보시겠어요??

**Reply by Yenowme** [reply]
node, 버전 업데이트하고 진행하니까 잘 작동하네요 ㅎㅎ 답변 감사합니다!

**Comment by minhoryang** [comment, marked as answer]
이 문제는 vue-app에서 `npm install`시에 Node 16을 사용하셔서 node-sass 설치과정에서 문제가 발생한 것입니다. 
- 해당 문제를 Node 16+에서도 회피하여 사용할 수 있지만,
- 해당 과제에서 Node LTS 버전을 사용해달라고 명시가 되어있고,
- 이 프로젝트를 그대로 Azure Static Web App에 빌드를 요청해도 별 문제없이 설치가 되는 것을 확인하실 수 있기에,
- Node 14 (Active LTS) 를 사용하셔야합니다.

nodenv나 nvm등을 이용해, node 14버전으로 옮겨서 작업해주시는게 좋을 것 같습니다.

**Reply by minhoryang** [reply]
About [Node LTS (long-term support)](https://nodejs.org/en/about/releases/)

**Reply by Yenowme** [reply]
node, 버전 업데이트하고 진행하니까 잘 작동하네요 ㅎㅎ 답변 감사합니다!

**Reply by minhoryang** [reply]
Answered로 해주시면 다른분들에게도 도움이 될 겁니다 :)

**Reply by Yenowme** [reply]
명확한 설명 감사합니다 ㅎㅎ

**Reply by minhoryang** [reply]
아이고 Answered로 하면, 답변 수정이 안되나보네요. 몇가지 링크를 더 달아볼까했는데 ㅠㅠ

**Reply by Yenowme** [reply]
앗 그럼 잠시 unmark 해드릴까요?? (했습니다! 완료되시면 다시 말씀해주세요 ㅎㅎ)

**Reply by minhoryang** [reply]
@Yenowme 지금 수정했는데도, 따로 업데이트가 안되네요ㅎㅎ 버그인가봐요. Github쪽에 따로 리포트해서 해결해보겠습니다. 다시 Answered로 해주세요. 감사합니다!
